src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -741,11 +741,18 @@ void MainWindow::update_workspace()
     update_pause_resume_checkbox(false);
     m_ui->attribute_editor_scrollarea_contents->setEnabled(true);
 
-    // Add/remove light paths tab.
+    // Enable/disable light paths
     if (m_project_manager.is_project_open() &&
         m_project_manager.get_project()->get_light_path_recorder().get_light_path_count() > 0)
-        add_light_paths_tab();
-    else remove_light_paths_tab();
+    {
+        for (const_each<ViewportTabCollection> i = m_viewport_tabs; i; ++i)
```

Comment:
`foundation::const_each<>` is deprecated and will be removed. Use range-based `for` loop in new code.